### PR TITLE
nsis: update to 3.09

### PR DESCRIPTION
**Pull request by idleberg**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by reneeotten** [comment]
@idleberg this fails on the CI - likely to do with the patches. Can  you take another look please?

**Comment by idleberg** [comment]
I‘m not entirely sure how to interprete the log of the failed pipeline (on a mobile phone that is). Did the final attempt to download both files succeed?

> DEBUG: Fetching distfile failed: The requested URL returned error: 404

Before taking a deeper look: do you think you could trigger the pipeline again to see if the problem resolves itself? In my experience those SourceForge mirrors can be unreliable on occasion.

**Comment by ryandesign** [comment]
CI on macOS 13 succeeded, failed on macOS 11 with this error:

```
patching file Docs/src/bin/halibut/misc.c
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file Docs/src/bin/halibut/misc.c.rej
Command failed:  cd "/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_nsis/nsis/work/nsis-3.09-src" && /usr/bin/patch -p0 < '/Users/runner/work/macports-ports/macports-ports/ports/devel/nsis/files/patch-halibut_for_osx_pre10_15.diff'
Exit code: 1
Error: Failed to patch nsis: command execution failed
```

and was canceled on macOS 12 since it failed on macOS 11.

Rerunning CI won't change this outcome.

The build success on macOS 13 is a surprise; if the patch failed to apply that should have been reported. But macOS 13 has a different patch implementation than macOS 12 and earlier and we are still becoming acquainted with what it does in various situations. Maybe it silently accepts patches that were already applied. The macOS 13 CI log has aged out by now so I can't check what it said, but the fix is pretty clear: Remove the patch since it's no longer needed.

**Comment by idleberg** [comment]
Thank you, @ryandesign. I followed your advice and now CI succeeds for all macOS versions.

File: devel/nsis/Portfile [hunk 1]
```diff
@@ -47,8 +47,6 @@ post-extract {
     system -W ${workpath} "unzip ${distpath}/nsis-${version}.zip"
 }
 
-patchfiles          patch-halibut_for_osx_pre10_15.diff
```
Comment (reneeotten) [inline devel/nsis/Portfile:50]: You should also remove the actual patchfile.

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by herbygillot** [comment]
This is building now and ready to merge, but 2 more minor requests.

@idleberg, can you:

1) Remove the `platforms darwin` line? This is now implicit in MacPorts and no longer needs to be present in Portfiles.

2) Make the homepage URL HTTPS? ( `homepage            https://n....`)

I think once we do that this should be good to go. Thanks.

**Pull request merged**